lib/stoplight/data_store/redis.rb

```diff
@@ -93,6 +96,17 @@ def with_notification_lock(light, from_color, to_color)
 
       private
 
+      # @param light [Stoplight::Light]
+      # @param time [Time]
+      def remove_outdated_failures(light, time, transaction: @redis)
+        failures_key = failures_key(light)
+
+        # Remove all errors happened before the window start
+        transaction.zremrangebyscore(failures_key, 0, time.to_i - light.window_size)
+        # Keep at most +light.threshold+ number of errors
+        transaction.zremrangebyrank(failures_key, 0, -light.threshold - 1)
```

Comment:
Why do we want to keep at most `light.threshold` errors?

Reply:
I added a class-level comment describing this. Long story short: Since we use a threshold to trigger notifications, it does not matter if we have more failures than the threshold value. So every time we save a new failure, we ensure we don't store old failures that could occupy a lot of Redis space and put pressure on the `get_failures` method.